app/controllers/shared/api.php

```diff
@@ -709,10 +723,11 @@
         }
 
         if (!$user->isEmpty()) {
+            $user->setAttribute('type', Auth::ACTIVITY_TYPE_USER);
             $queueForAudits->setUser($user);
         }
 
-        if (!empty($queueForAudits->getResource()) && !empty($queueForAudits->getUser()->getId())) {
+        if (!empty($queueForAudits->getResource()) && !$queueForAudits->getUser()->isEmpty()) {
```

Comment:
For cases where the request is not made by API key or user, the document from `getUser()` will be empty. That means we won't track it, right? Should we?

Reply:
Yes. Added an `Anonymous` type to audits.